AdyenComponents/UPI/UPIComponent.swift

```diff
@@ -0,0 +1,200 @@
+        let item = FormLabelItem(text: localizedString(.UPIQrcodeGenerationMessage, configuration.localizationParameters),
+                                 style: configuration.style.footnoteLabel)
+        item.style.textAlignment = .center
+        item.identifier = ViewIdentifierBuilder.build(scopeInstance: self,
+                                                      postfix: ViewIdentifier.generateQRCodeContainerItem)
+        return FormContainerItem(content: item.addingDefaultMargins())
+    }()
+
+    /// The QRCode generation message view.
+    private lazy var qrCodeGenerationImageItem: FormImageItem = {
+        let imageView = FormImageItem(name: "qrcode")
+        imageView.isHidden.wrappedValue = true
+        return imageView
+    }()
+
+    /// The continue button item.
+    internal lazy var continueButton: FormButtonItem = {
+        let item = FormButtonItem(style: configuration.style.mainButtonItem)
+        item.identifier = ViewIdentifierBuilder.build(scopeInstance: self,
+                                                      postfix: ViewIdentifier.continueButtonItem)
+        item.title = localizedString(.continueTitle, configuration.localizationParameters)
+        item.buttonSelectionHandler = { [weak self] in
+            self?.didSelectContinueButton()
+        }
+        return item
+    }()
+
+    // MARK: - Private
+
+    private lazy var formViewController: FormViewController = {
+        let formViewController = FormViewController(style: configuration.style)
+        formViewController.localizationParameters = configuration.localizationParameters
+        formViewController.title = paymentMethod.displayInformation(using: configuration.localizationParameters).title
+        formViewController.append(FormSpacerItem(numberOfSpaces: 1))
+        formViewController.append(instructionsLabelItem.addingDefaultMargins())
+        formViewController.append(FormSpacerItem(numberOfSpaces: 1))
+        formViewController.append(upiFlowSelectionItem.addingDefaultMargins())
+        formViewController.append(qrCodeGenerationImageItem)
+        qrCodeGenerationLabelContainerItem.isHidden.wrappedValue = true
+        formViewController.append(FormSpacerItem(numberOfSpaces: 1))
+        formViewController.append(qrCodeGenerationLabelContainerItem)
+        formViewController.append(virtualPaymentAddressItem)
+        formViewController.append(FormSpacerItem(numberOfSpaces: 2))
+        formViewController.append(continueButton)
+
+        return formViewController
+    }()
+
+    private func didSelectContinueButton() {
+        guard formViewController.validate() else { return }
+
+        continueButton.showsActivityIndicator = true
+        formViewController.view.isUserInteractionEnabled = false
+
+        switch currentSelectedIndex {
+        case 0:
+            if !virtualPaymentAddressItem.value.isEmpty {
+                let details = UPIComponentDetails(type: "upi_collect",
```

Comment:
Maybe put this one to `Constants` ? 🤔

Reply:
Fixed. Here is the commit: 32a3cdc9cc0d9e1ff71bdd9886176033095764dd